### ISSUE TITLE
Implementation mismatch

### ISSUE THREAD

**Issue by blatr** [fmu2/realNVP#4]
Hi. 

What's the purpose of [concating](https://github.com/fmu2/realNVP/blob/8d36691df215af3678440ccb7c01a13d2b441a4a/realnvp.py#L316) x, -x and mask in CheckerboardAffineCoupling layer ?
`x_ = self.in_bn(x * mask)    
x_ = torch.cat((x_, -x_), dim=1)    
x_ = torch.cat((x_, mask), dim=1)`
I havent't managed to find any references for this operation in original paper.
